components/Forms/CreateResourceForm/CreateResourceForm.js

```diff
@@ -0,0 +1,149 @@
+  onSuccess: func.isRequired,
+  onFailure: func.isRequired,
+};
+
+function CreateResourceForm({ categories, languages, onSuccess, onFailure }) {
+  const initialValues = React.useMemo(() => validationSchema.cast(), []);
+
+  const categoryOptions = categories.map(({ name }) => ({
+    value: name.toLowerCase(),
+    label: name,
+  }));
+
+  const languageOptions = languages.map(({ name }) => ({
+    value: name.toLowerCase(),
+    label: name,
+  }));
+
+  const handleSubmit = async ({ cost, ...rest }, actions) => {
+    const values = { ...rest, free: cost === free.value };
+
+    try {
+      await createResource(values);
+
+      actions.setSubmitting(false);
+      actions.resetForm();
+
+      onSuccess();
+    } catch (error) {
+      actions.setSubmitting(false);
+
+      onFailure(getServerErrorMessage(error));
+    }
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={handleSubmit}
+      validationSchema={validationSchema}
+    >
+      {({ isSubmitting }) => (
+        <Form className={styles.CreateResourceForm}>
+          <Field
+            type="text"
+            name="name"
+            label="Name*"
+            placeholder="What is this resource called?"
+            component={Input}
+            disabled={isSubmitting}
+            required
+          />
+
+          <Field
+            type="url"
+            name="url"
+            label="URL*"
+            placeholder="Where can one access this resource?"
+            component={Input}
+            disabled={isSubmitting}
+            required
+          />
+
+          <Field
+            name="category"
+            label="Category*"
+            placeholder="What category best applies to this resource?"
+            options={categoryOptions}
+            component={Select}
+            isDisabled={isSubmitting}
+            required
+          />
+
+          <Field
+            name="languages"
+            label="Programming language(s)"
+            placeholder="What language or languages are involved?"
+            options={languageOptions}
+            component={Select}
+            isDisabled={isSubmitting}
+            required
+          />
+
+          <Field
+            name="cost"
+            label="Cost*"
+            placeholder="Is this resource free or paid?"
+            options={costOptions}
+            component={Select}
+            isDisabled={isSubmitting}
+            required
+          />
+
+          <Field
+            type="text"
+            name="notes"
+            label="Notes"
+            placeholder="Anything worth noting regarding this resource?"
+            component={Input}
+            disabled={isSubmitting}
+            required
+          />
+
+          <Button
+            className={styles.submitButton}
+            type="submit"
+            theme="secondary"
+            disabled={isSubmitting}
+          >
+            Submit
+          </Button>
```

Comment:
In addition to the submit button, we should also have a "back" or "X" or "cancel" button, in case the user decides against submitting their resource